### ISSUE TITLE
Extend People Profiles

### ISSUE THREAD

**Issue by xmatthewx** [mozilla/network-pulse#604]
Pulse has lots of interesting people – project creators and people connectors. This epic aims to surface these people, extend discovery, and allow users to share information about themselves. 

 - Entry includes creators and submitter
 - Names link to profile view when available
 - Profile view lists person's related entries (created, submitted, fav'd)
 - User has means to edit profile fields (design pending)
 - Pulse has notion of mini vs full profile, full profile unlocked by moderator
 - Adding a creator offers autocomplete with existing people

**Comment by jessevondoom** [comment]
Love the mini v full concept.

**Comment by xmatthewx** [comment]
@alanmoo to file tickets and plan sprint and kickoff.

**Comment by xmatthewx** [comment]
@alanmoo – I didn't have a chance to create sub-epics to organize this work a bit more. 

I'm thinking these might make sense:
 - Enable users to Create a Profile
 - Connect Profiles to Entries and Creators
 - Extend Profiles for Mozilla programs

And... 

Here is [a fancy table](https://docs.google.com/presentation/d/12o0P9wWGlUEuY-247fWuHluVQsdfm9Htqlfuo8OQUio/edit?ts=5996143c#slide=id.g24e286cd1a_0_7) I reviewed with the team to clarify cases for when a Profile is public or not. We had 98%-ish agreement. @kristinashu can own this if questions pop up.

<img width="1147" alt="screen shot 2017-08-18 at 5 30 59 pm" src="https://user-images.githubusercontent.com/53411/29478470-0f36342e-843b-11e7-8134-09186ce3ca18.png">

**Comment by alanmoo** [comment]
One clarification: For the second to last row (User without public profile), is it correct to assume that we're not linking to the user's profile in the front end, but we do retain that connection in the back end so that once a profile does exist, we can link to it?

**Comment by kristinashu** [comment]
Yes, definitely retain the connection in the back end. The front end is where we had our 2% of disagreement... We are unsure if people should have to have a public profile (just basic profile e.g. just their user name) in order to post an entry and if so, when should a user need to have a public profile (when they first sign up vs after they fill in their post entry). 

@hannahkane is that how you would summarize it? Can we discuss and finalize this part once Matthew is back next week?

**Comment by xmatthewx** [comment]
Great progress. I need to work with @alanmoo to break up remain tasks into a few epics.

**Comment by Pomax** [comment]
profile routing:
1. client will expose a `/myprofile` which is for editing your own profile
2. client will expose a `/profiles/:id` which is for viewing public profiles for any user
3. client will expose a `/profiles/:me` which as alias for viewing "my own" public profile. The only difference between /:id and /me is that there will be an "edit" link to take the user to /myprofile

**Comment by xmatthewx** [comment]
I've started a draft list of acceptance tests to help us clarify things we need done before we can ship profiles to users. We can hack on this during grooming: https://public.etherpad-mozilla.org/p/pulse-profiles-checklist 

cc @kristinashu

**Comment by xmatthewx** [comment]
We enter MozFest ready to tell dynamic stories rooted in profiles and projects, in realtime, with strong context on the Internet Health movement….so that people exit MozFest with ways to connect beyond the event. 
- people can post profiles/projects
- editors can link to people/project in medium posts featured on site

**Comment by xmatthewx** [comment]
Hi team. @kristinashu proposed that we simplify Profiles and get rid of the notion of Public/Not-yet-public. Like Twitter and other platforms, creating an account includes creating a profile. People are of course free to browse without an account. 

To this end, we get simpler logic, fewer engineering twists, and a cleaner experience. We will need to reach out to current users (mostly staff), and give them a chance to remove their account.

In short, users will tap login, see a new /login view (tbd) with terms and a button for g-auth. 

![image](https://user-images.githubusercontent.com/53411/30187055-108f5a92-93f6-11e7-8e0d-6ceb8fcd34c9.png)
comment below or see [the table here](https://docs.google.com/presentation/d/12o0P9wWGlUEuY-247fWuHluVQsdfm9Htqlfuo8OQUio/edit?ts=59b1bb59#slide=id.g25311d4f33_0_0)

**Comment by alanmoo** [comment]
Makes sense to me. How does this impact existing tickets?

**Comment by xmatthewx** [comment]
My sense is that this layer of detail hasn't yet been implemented, just discussed. We can check with team at end of stand today.

**Comment by alanmoo** [comment]
@xmatthewx We could use some feedback from you around what work is still needed before Mozfest

**Comment by xmatthewx** [comment]
At this point, I think we need to huddle up and review what's needed to ship the most basic, but complete experience. Kristina and I have started a deck to review with the team.

**Comment by xmatthewx** [comment]
@kristinashu kindly phased our final login+profile work. Below is a summary as detailed in [the deck](https://docs.google.com/presentation/d/12o0P9wWGlUEuY-247fWuHluVQsdfm9Htqlfuo8OQUio/edit#slide=id.g236892fde9_0_11). Please make sure there tickets to represent all of your work.

### P1 flow
 - Login anywhere > page with “log in with G-auth” button
 - Redirect
    - redirect to entry form if plus button was origin
    - else redirect to profile if never visited
    - else redirect to origin

### P2 flow
 - Same as above
 - adds notice to a couple views for users who have not yet seen /myprofile

### P3 Flow
 - Login anywhere > login view (new copy)
 - Redirect same as above (but to profile form, not profile)

**Comment by xmatthewx** [comment]
I think we can close this! Though I recommend taking a look at all the closed items linked to this ticket. Truly epic!
